wpengine/faustjs
# Have Faust CLI verify if WPGraphQL is active

## Tasks


## Description

These changes handle verifying if the WPGraphQL plugin is active in order to provide more helpful info to a Faust user. Prior to these changes, if WPGraphQL was unavailable, a [Next.js pre-render error](https://nextjs.org/docs/messages/prerender-error) was thrown that wasn't too helpful.

These changes also refactor the internal structure of this package as the utils/ directory was getting a little challenging to visually search through.

These changes also setup Jest tests within the `@faustwp/cli` package. 🎉 

## Testing

1. Clone this branch: `cli-health-check`
2. Update the faustwp example project to use a test WordPress site
3. Deactivate WPGraphQL from your connected WordPress site
4. Run `npm run build -w @faustwp/getting-started-example`
5. Observe that the build command will warn user and is non-blocking (see screenshot below - note that the error is now a warning)
6. Re-activate WPGraphQL
7. Run `npm run build -w @faustwp/getting-started-example`
8. Observe that the build command was successful

## Screenshots

<img width="976" alt="Screenshot 2023-03-08 at 2 01 26 PM" src="https://user-images.githubusercontent.com/6676674/223815126-217f3000-fe31-44a6-9812-fd9b0de55883.png">
